airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/CdcWalLogsPostgresSourceDatatypeTest.java

```diff
@@ -0,0 +1,122 @@
+        .put("method", "CDC")
+        .put("replication_slot", SLOT_NAME_BASE)
+        .put("publication", PUBLICATION)
+        .put("initial_waiting_seconds", INITIAL_WAITING_SECONDS)
+        .build());
+    config = Jsons.jsonNode(ImmutableMap.builder()
+        .put(JdbcUtils.HOST_KEY, HostPortResolver.resolveHost(container))
+        .put(JdbcUtils.PORT_KEY, HostPortResolver.resolvePort(container))
+        .put(JdbcUtils.DATABASE_KEY, container.getDatabaseName())
+        .put(JdbcUtils.SCHEMAS_KEY, List.of(SCHEMA_NAME))
+        .put(JdbcUtils.USERNAME_KEY, container.getUsername())
+        .put(JdbcUtils.PASSWORD_KEY, container.getPassword())
+        .put("replication_method", replicationMethod)
+        .put("is_test", true)
+        .put(JdbcUtils.SSL_KEY, false)
+        .put("snapshot_mode", "never")
+        .build());
+
+    dslContext = DSLContextFactory.create(
+        config.get(JdbcUtils.USERNAME_KEY).asText(),
+        config.get(JdbcUtils.PASSWORD_KEY).asText(),
+        DatabaseDriver.POSTGRESQL.getDriverClassName(),
+        String.format(DatabaseDriver.POSTGRESQL.getUrlFormatString(),
+            container.getHost(),
+            container.getFirstMappedPort(),
+            config.get(JdbcUtils.DATABASE_KEY).asText()),
+        SQLDialect.POSTGRES);
+    final Database database = new Database(dslContext);
+
+    database.query(ctx -> {
+      ctx.execute(
+          "SELECT pg_create_logical_replication_slot('" + SLOT_NAME_BASE + "', 'pgoutput');");
+      ctx.execute("CREATE PUBLICATION " + PUBLICATION + " FOR ALL TABLES;");
+      ctx.execute("CREATE EXTENSION hstore;");
+      return null;
+    });
+
+    database.query(ctx -> ctx.fetch("CREATE SCHEMA TEST;"));
+    database.query(ctx -> ctx.fetch("CREATE TYPE mood AS ENUM ('sad', 'ok', 'happy');"));
+    database.query(ctx -> ctx.fetch("CREATE TYPE inventory_item AS (\n"
+        + "    name            text,\n"
+        + "    supplier_id     integer,\n"
+        + "    price           numeric\n"
+        + ");"));
+
+    database.query(ctx -> ctx.fetch("SET TIMEZONE TO 'MST'"));
+    return database;
+  }
+
+  @Override
+  protected void tearDown(final TestDestinationEnv testEnv) {
+    dslContext.close();
+    container.close();
+  }
+
+  public boolean testCatalog() {
+    return true;
+  }
+
+  @Override
+  protected void addTimeWithTimeZoneTest() {
+    // time with time zone
+    for (final String fullSourceType : Set.of("timetz", "time with time zone")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType("timetz")
+              .fullSourceDataType(fullSourceType)
+              .airbyteType(JsonSchemaType.STRING_TIME_WITH_TIMEZONE)
+              .addInsertValues("null", "'13:00:01'", "'13:00:00+8'", "'13:00:03-8'", "'13:00:04Z'", "'13:00:05.012345Z+8'", "'13:00:06.00000Z-8'")
+              // A time value without time zone will use the time zone set on the database, which is Z-7,
```

Comment:
I think this comment is inaccurate? (since everything is returned in UTC)

Reply:
I think its relevant cause it helps us understand why `13:00:01` is being converted to `20:00:01.000000Z` cause since  `13:00:01` doesnt have timezone, we use the DB's timezone which is -7 and thus the value becomes `13:00:01-07` which in UTC representation would be  `20:00:01.000000Z`